locator/abstract_replication_strategy.cc

```diff
@@ -308,10 +308,20 @@ abstract_replication_strategy::get_pending_address_ranges(const token_metadata_p
 
 future<mutable_effective_replication_map_ptr> calculate_effective_replication_map(abstract_replication_strategy::ptr_type rs, token_metadata_ptr tmptr) {
     replication_map replication_map;
-
-    for (const auto &t : tmptr->sorted_tokens()) {
-        auto eps = co_await rs->calculate_natural_endpoints(t, *tmptr);
-        replication_map.emplace(t, eps.get_vector());
+    const auto& sorted_tokens = tmptr->sorted_tokens();
+
+    replication_map.reserve(sorted_tokens.size());
+    if (rs->has_static_natural_endpoints()) {
+        auto eps = co_await rs->calculate_natural_endpoints(sorted_tokens.front(), *tmptr);
```

Comment:
The name `has_static_natural_endpoints` tells me the token->endpoint mapping never changes, not that all tokens map to the same endpoint-set.

Reply:
whatvwould be a better name? uniform?

Reply:
token_independent?

Reply:
a bit long, but fine :)

Reply:
How about `per_token_natural_endpoints`?

Reply:
> How about `per_token_natural_endpoints`?

I cannot decipher this one.

Reply:
Then how about `natural_endpoints_depend_on_token`? 
I'm just trying to find something in simple English... :)

Reply:
This sounds good.

Reply:
yes